dev/AppNotifications/AppNotificationManager.cpp

```diff
@@ -167,6 +168,28 @@ namespace winrt::Microsoft::Windows::AppNotifications::implementation
         }
     }
 
+    void AppNotificationManager::Register()
+    {
+        RegisterHelper(winrt::hstring{} /* displayName */, std::wstring{} /* iconFilePath */);
+    }
+
+    void AppNotificationManager::Register(hstring const& displayName, winrt::Uri const& iconUri)
+    {
+        THROW_HR_IF_MSG(E_ILLEGAL_METHOD_CALL, AppModel::Identity::IsPackagedProcess(), "Not applicable for packaged applications");
+
+        THROW_HR_IF(E_INVALIDARG, (displayName != winrt::hstring{}) xor (iconUri != nullptr));
+
+        std::wstring iconFilePath{};
+        if (iconUri != nullptr)
+        {
+            iconFilePath = iconUri.Path();
+            iconFilePath = iconFilePath.substr(1, iconFilePath.size() - 1);
+            winrt::check_bool(std::filesystem::exists(std::filesystem::path{ iconFilePath }));
```

Comment:
You are not checking if the extension is supported or not here.

Reply:
@danielayala94 You said there was a helper for this?

Reply:
I went all the way to `RegisterAssetsHelper` to verify that the file extension isn't being checked.

@sharath2727 There is an inline helper in `ShellLocalization.cpp` -> `IsIconFileExtensionSupported`. Declare it in the header so you can consume it in `AppNotificationManager`.